### PR TITLE
Improve PacketFence Developer's Guide

### PR DESCRIPTION
**Pull request by nqb**
# Description
Improve PacketFence Developer's Guide

# Impacts
- PacketFence Developer's Guide

# Delete branch after merge
YES

### CODE REVIEW

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 1]
```diff
@@ -118,12 +118,44 @@ HTTP JSON API
 
 PacketFence provides an HTTP JSON API which exposes most of its functionality. 
 
-The full API spec documentation for the latest stable release is available from: http://petstore.swagger.io/?url=https://raw.githubusercontent.com/inverse-inc/packetfence/stable/docs/api/spec/openapi.yaml 
+The full API spec documentation for :
```
Comment (cgx) [inline docs/PacketFence_Developers_Guide.asciidoc]: In English, no space is placed before a colon.

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 2]
```diff
@@ -118,12 +118,44 @@ HTTP JSON API
...
+
+* the latest stable release is available from: https://packetfence.org/doc/api/
+* the latest devel release is available from: https://packetfence.org/doc/api-devel/
 
 The API is exposed on 127.0.0.1 (localhost) as well as the management interface of the server for remote calls.
+You should use 9999 TCP port and a secure connection (HTTPS) to ask it.
```
Comment (cgx) [inline docs/PacketFence_Developers_Guide.asciidoc]: I propose: *Users access the API on TCP port 9999 over a secure connection (HTTPS).*

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 3]
```diff
@@ -118,12 +118,44 @@ HTTP JSON API
...
 
 Authentication can be done using either the webservices credentials or any credentials that are valid on the web admin interface.
 
+How to use API
```
Comment (cgx) [inline docs/PacketFence_Developers_Guide.asciidoc]: *How to use the API*

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 4]
```diff
@@ -118,12 +118,44 @@ HTTP JSON API
...
+~~~~~~~~~~~~~~
+
+NOTE: You can use API spec documentation mentionned above to have ready to use curl commands with "Try it out" button.
```
Comment (cgx) [inline docs/PacketFence_Developers_Guide.asciidoc]: *NOTE: The API spec documentation mentioned above provides ready-to-use curl commands through the "Try it out" button.*

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 5]
```diff
@@ -118,12 +118,44 @@ HTTP JSON API
...
+ --insecure |
+ python -m json.tool
+
+You will get following response :
+
+ {
+    "token": "MY_TOKEN"
+ }
+
+Then use this token to ask API (here get domains config) :
```
Comment (cgx) [inline docs/PacketFence_Developers_Guide.asciidoc]: I propose: *Then use this token to query the API. The following example fetches the domains configuration:*

**Review by cgx — CHANGES_REQUESTED** [review]

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 6]
```diff
@@ -118,12 +118,44 @@ HTTP JSON API
...
+You should use 9999 TCP port and a secure connection (HTTPS) to ask it.
 
 Authentication can be done using either the webservices credentials or any credentials that are valid on the web admin interface.
 
+How to use API
+~~~~~~~~~~~~~~
+
+NOTE: You can use API spec documentation mentionned above to have ready to use curl commands with "Try it out" button.
+
+First, get an authentication token with webservices credentials or an admin account:
```
Comment (julsemaan) [inline docs/PacketFence_Developers_Guide.asciidoc]: with the webservices credentials

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 7]
```diff
@@ -118,12 +118,44 @@ HTTP JSON API
 
 PacketFence provides an HTTP JSON API which exposes most of its functionality. 
 
-The full API spec documentation for the latest stable release is available from: http://petstore.swagger.io/?url=https://raw.githubusercontent.com/inverse-inc/packetfence/stable/docs/api/spec/openapi.yaml 
+The full API spec documentation for:
```
Comment (cgx) [inline docs/PacketFence_Developers_Guide.asciidoc]: I propose *The full API specification for:*

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 8]
```diff
@@ -118,12 +118,44 @@ HTTP JSON API
...
+* the latest stable release is available from: https://packetfence.org/doc/api/
+* the latest devel release is available from: https://packetfence.org/doc/api-devel/
 
 The API is exposed on 127.0.0.1 (localhost) as well as the management interface of the server for remote calls.
+Users access the API on TCP port 9999 over a secure connection (HTTPS).
 
 Authentication can be done using either the webservices credentials or any credentials that are valid on the web admin interface.
 
+How to use the API
+~~~~~~~~~~~~~~
```
Comment (cgx) [inline docs/PacketFence_Developers_Guide.asciidoc]: Add a few more `~`.

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 9]
```diff
@@ -118,12 +118,44 @@ HTTP JSON API
...
+
+NOTE: The API spec documentation mentionned above provides ready-to-use curl commands through the "Try it out" button.
```
Comment (cgx) [inline docs/PacketFence_Developers_Guide.asciidoc]: I propose *The API specification mentionned* ...

**Review by cgx — CHANGES_REQUESTED** [review]

File: docs/PacketFence_Developers_Guide.asciidoc (hunk 8 above)
Comment (nqb) [reply docs/PacketFence_Developers_Guide.asciidoc]: Not necessary for me. Check other titles for example.

**Review by cgx — APPROVED** [review]

**Comment by nqb** [comment]
Do not merge for the moment, I would like to improve some other sections.

**Pull request merged**